File: frontend/src/app/dtos/ticket.ts
import {UserProfile} from './user-profile';
import {Show} from './show';
import {SeatingChartSeat, SeatingChartSector} from './seating-chart';
import {Status} from 'tslint/lib/runner';
import {Order} from './order';

export interface Ticket {

        ticketId: number;
        createdAt: string;
        user: UserProfile;
        show: Show;
        sector: SeatingChartSector;
        seat?: SeatingChartSeat;
        price: number;
        status: Status;
        reservationNumber: string;
        ticketOrder: Order;
}

File: backend/src/main/java/at/ac/tuwien/sepm/groupphase/backend/domain/model/Ticket.java
package at.ac.tuwien.sepm.groupphase.backend.domain.model;

import lombok.Builder;
import lombok.Value;

import java.time.Instant;

@Value
@Builder(builderClassName = "Builder", toBuilder = true)
public class Ticket {
    Long ticketId;
    Instant createdAt;
    ApplicationUser user;
    Show show;
    Sector sector;
    Seat seat;
    Double price;
    TicketStatus status;
    String reservationNumber;
    Order ticketOrder;
}
